lib/datasource/docker/index.ts

```diff
@@ -420,33 +420,41 @@ async function getTags(
   }
 }
 
+export function getConfigResponseBeforeRedirectHook(options: any): void {
+  if (options.search && options.search.indexOf('X-Amz-Algorithm') !== -1) {
```

Comment:
```suggestion
  if (options.search?.indexOf('X-Amz-Algorithm') !== -1) {
```
Simplify condition

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining

Reply:
btw while we're at it, I prefer the readability if `.includes('foo')` than `.indexOf('foo') !== -1`

Reply:
voila...